app/controllers/application_controller.rb

```diff
@@ -296,6 +287,21 @@ def authenticated?(reason_params)
     return true
   end
 
+  def set_post_redirect_and_headers(params, reason_params)
```

Comment:
This extraction can be dropped given we won't be sharing it between 2 methods now.